app/main.py

```diff
@@ -0,0 +1,198 @@
+    def amqp_params(services):
+        try:
+            uri = services["rabbitmq"][0]["credentials"]["protocols"]["amqp"]["uri"]
+        except (IndexError, KeyError):
+            uri = "amqp://{user}:{password}@{hostname}:{port}/{vhost}".format(
+                hostname=os.getenv("SEFT_RABBITMQ_HOST", "localhost"),
+                port=os.getenv("SEFT_RABBITMQ_PORT", 5672),
+                user=os.getenv("SEFT_RABBITMQ_DEFAULT_USER", "rabbit"),
+                password=os.getenv("SEFT_RABBITMQ_DEFAULT_PASS", "rabbit"),
+                vhost=os.getenv("SEFT_RABBITMQ_DEFAULT_VHOST", "%2f")
+            )
+        return {
+            "amqp_url": uri,
+            "queue_name": "Seft.Responses",
+        }
+
+    @staticmethod
+    def encrypt_params(services, locn="."):
+        log = logging.getLogger("sdx-seft-publisher")
+        pub_fp = os.getenv("RAS_SEFT_PUBLIC_KEY", os.path.join(locn, "test_no_password.pub"))
+        priv_fp = os.getenv("SDX_SEFT_PRIVATE_KEY", os.path.join(locn, "test_no_password.pem"))
+        priv_key = None
+        pub_key = None
+        try:
+            with open(priv_fp, "r") as key_file:
+                priv_key = key_file.read()
+        except Exception as e:
+            log.warning("Could not read key {0}".format(priv_fp))
+            log.warning(e)
+        try:
+            with open(pub_fp, "r") as key_file:
+                pub_key = key_file.read()
+        except Exception as e:
+            log.warning("Could not read key {0}".format(pub_fp))
+            log.warning(e)
+
+        rv = {
+            "public_key": pub_key,
+            "private_key": priv_key,
+            "private_key_password": os.getenv("SDX_SEFT_PRIVATE_KEY_PASSWORD", None),
+        }
+        return rv
+
+    @staticmethod
+    def ftp_params(services):
+        return {
+            "user": os.getenv("SEFT_FTP_USER", "user"),
+            "password": os.getenv("SEFT_FTP_PASS", "password"),
+            "host": os.getenv("SEFT_FTP_HOST", "127.0.0.1"),
+            "port": os.getenv("SEFT_FTP_PORT", 2121),
+        }
+
+    def __init__(self, args, services):
+        self.args = args
+        self.services = services
+        self.publisher = DurableTopicPublisher(**self.amqp_params(services))
+
+    def transfer_files(self):
+        log = logging.getLogger("sdx-seft-publisher")
+        if not self.publisher.publishing:
+            log.warning("Publisher is not ready.")
+            return
+
+        log.info("Looking for files...")
+        worker = FTPWorker(**self.ftp_params(self.services))
+        encrypter = Encrypter(**self.encrypt_params(self.services, locn=self.args.keys))
+        with worker as active:
+            if not active:
+                return
+
+            for job in active.get(active.filenames):
+                if job.fn not in self.recent:
+                    data = job._asdict()
+                    data["contents"] = base64.standard_b64encode(job.contents).decode("ascii")
```

Comment:
Rather than "contents" it should be "file"
We also need a "filename" attribute with the name of the file.